internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -630,6 +648,17 @@ func (c *Coordinator) compute() (map[string]interface{}, []component.Component,
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to convert ast to map[string]interface{}: %w", err)
 	}
+
+	if c.monitorMgr.Enabled() {
+		componentIDToBinary, err := c.specs.ToComponentIdsInputMap(cfg)
```

Comment:
This is duplicating a lot of work in rendering the components model just to get the mapping of components to input binaries.

I have a PR that makes even more changes to `ToComponents` because of the shipper enablement. So by splitting this out will mean both need to be updated and kept in sync.

This should be changed to make the mapping by looking at the generated model, instead of computing the model itself. Let `ToComponents` to its job then use that model to get the mapping.

Reply:
After looking at the code more I can see this is done because basically the model has to be generated twice.

1. Generate the model.
2. Inject the configuration.
3. Re-generate the model.

If we need to keep it this way they I still believe that `ToComponents` should just be called twice. But I would prefer to see the code inject the configuration changes directly into the components model instead of placing it into the configuration and then re-generating the model.

I also wonder how this works with the `inspect` command? Maybe we should just inject the monitoring code directly into the `ToComponents`. That way its very simple and easy to take a configuration and turn it into a model with monitoring enabled.

Reply:
i like injecting approach, will change it like that. otherwise yes 2 rounds need to be run.

Reply:
also a note why i would prefer it to have it inject into a config rather than components. 
components is an internal representations resulting from a configuration already being a bit complex, which is shared across products. so I see it's less likely to change, meaning it will make future changes to components model easier with lower risk of introducing bugs into monitoring. 

also ToComponents checks for input being enabled and output supported on a platform, these checks would need to be performed upon injection as well, so we would end up with similar code duplication.

if you stronly object i can redo this but I would rather keep the concerns separate